### DISCUSSION TITLE
Importer

### DISCUSSION THREAD

**Discussion by Serph91P** [firefly-iii/firefly-iii#7843] [Firefly III] [answered]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [ ] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

The Importer imports Data from Nordigen just fine. My problem is with the data that gets imported.  This is a fresh install because I thought I did something wrong and read the instructions again after I deleted everything. I did just as the documentation says to only import 10 Transactions at a time.  Almost all of those imported transactions are not correctly mapped. The kinda default to one Destination Account only some are mapped correctly. I looked in the data on Nordigen directly. They are perfectly Fine. To me it seems like the creditorName is not looked at correctly. One of those Transaktion is actually from AMZN Mktp DE.1V15Y2XR4. Im based in EU/Germany, the bank used is DKB.
![grafik](https://github.com/firefly-iii/firefly-iii/assets/8904984/71837533-e63b-4dd2-82e8-5737fa82d659)

 If there is already and Issue open with this Iam sorry I did not find anything related.

### Debug information

Debug information generated at 2023-08-10 19:35:38 Europe/Berlin for Firefly III version **6.0.19**.

| Scope | Version |
| --- | --- |
| Firefly III | 6.0.19 |
| Firefly III API | 2.0.4 |
| Build | 553, 28-07-2023 14:22:29 UTC |
| Base Build | 13, 26-07-2023 07:34:31 CEST |
| DB version | 19 (exp. 19) |
| PHP | `8.2.8` |
| Host | `Linux` |

| System info | Value |
| --- | --- |
| System TZ | Europe/Berlin |
| Browser TZ | Europe/Oslo |
| App environment | local |
| App debug mode | false |
| App cache driver | file |
| App logging | debug, stack |
| Display errors | Off |
| Error reporting | ALL errors |
| Max upload | 67108864 (64 MB) |
| Interface | apache2handler |
| Default language | en_US |
| Default locale | equal |
| BCscale | 12 |
| DB drivers | mysql, pgsql, sqlite |
| Current driver | mysql |
| Trusted proxies (.env) | `` |

| User info | Value |
| --- | --- |
| Login provider | eloquent  |
| User guard | web  |
| Headers | N/A, N/A  |
| Stateful domains |  |
| Session start | 2023-07-01 00:00:00 |
| Session end | 2023-07-31 23:59:59 |
| Session first | 2023-08-10 19:17:27 |
| User ID | 1 |
| User language | en_US |
| User locale | en_US |
| Attempt at "en_US.utf8" | 'en_US.utf8' |
| Attempt at "en_US.UTF-8" | 'en_US.UTF-8' |
| User agent | Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/116.0 |

### Expected behaviour

Imported Transactions are shown correctly and have not to be edited after the import.

### Steps to reproduce

1. use imported to import Bank DKB Data via. Nordigen
2. Look at the imported data.

**Comment by JC5** [comment]
> To me it seems like the creditorName is not looked at correctly.

So, what did you expect to happen to account `AMZN Mktp DE.1V15Y2XR4`?

**Reply by Serph91P** [reply]
That the correct one is applied only one transaction is actually from Amazon the rest are completely different accounts.

**Reply by JC5** [reply]
And how should the importer know the difference between those accounts?

**Reply by Serph91P** [reply]
Sorry if I did not get this correctly, English is not my native language. So the importer is not be used to import my transactions from my bank account I have to do those manually? Because only one of those transactions is actually from Amazon the rest are PayPal or TÜV and so one they are completely different ones nothing with Amazon. I get that if they would be all from different Amazon accounts that it cannot differentiate those but the are from different accounts completely. 

Lesson learned will import all manually. Thank you for clarifying.

**Reply by JC5** [reply]
I'm not sure what you mean. The account names are taken directly from Nordigen. If they are not yet nice, you can use the mapping function from the importer or rules to clean them up.

**Reply by Serph91P** [reply]
Like I wrote before I checked in nordigen, there they are correct but after the import to firefly they are kinda copied like you see in the screenshot. The mapping function I don't seem to get because I can only chose the accounts and cannot create new mappings.

Thank you for your help

**Reply by Serph91P** [reply]
just that you get what I mean I got into Nordigen and copied the log below and removed my account number, those are the first transactrions from nordigen if you check them with the screen you will see what I mean.
 ```
{
    "transactions": {
        "booked": [
            {
                "entryReference": "REU2NTEyMDMwMDAwMTA2NTQ1NjU2NjI1LjAwMjAyMy0wOC0wOQ==",
                "bookingDate": "2023-08-09",
                "valueDate": "2023-08-09",
                "transactionAmount": {
                    "amount": "25.00",
                    "currency": "EUR"
                },
                "creditorAccount": {
                    "iban": "redacted"
                },
                "debtorName": "redacted",
                "debtorAccount": {
                    "iban": "DE10120300001083220390"
                },
                "remittanceInformationUnstructured": "Sie wissen schon",
                "internalTransactionId": "d3cdeeaa6f95651b6e8c5377df22d4b1"
            },
            {
                "entryReference": "REUwNDMwMDMwODgwMDcwMDA1NDAwNC05Ljk1MjAyMy0wOC0wOQ==",
                "endToEndId": "3NU3271RFKDTFWLE",
                "mandateId": "cZ)dORMgOEc-SLryhwUKpJ(TF:nY60",
                "creditorId": "DE31ZZZ00000563369",
                "bookingDate": "2023-08-09",
                "valueDate": "2023-08-09",
                "transactionAmount": {
                    "amount": "-9.95",
                    "currency": "EUR"
                },
                "creditorName": "AUDIBLE GMBH",
                "creditorAccount": {
                    "iban": "DE04300308800700054004"
                },
                "debtorAccount": {
                    "iban": "redacted"
                },
                "remittanceInformationUnstructured": "D01-1021357-1157436 Audible Gmbh 3NU3271RFKDTFWLE",
                "internalTransactionId": "41675952942a5af988d96a578af4fc9c"
            },
            {
                "entryReference": "REU5NjEyMDMwMDAwOTAwNTI5MDkwNC05LjQwMjAyMy0wOC0wOA==",
                "endToEndId": "483218666240328",
                "bookingDate": "2023-08-08",
                "valueDate": "2023-08-08",
                "transactionAmount": {
                    "amount": "-9.40",
                    "currency": "EUR"
                },
                "creditorName": "AMZN Mktp DE.1V15Y2XR4",
                "creditorAccount": {
                    "iban": "DE96120300009005290904"
                },
                "debtorAccount": {
                    "iban": "redacted"
                },
                "remittanceInformationUnstructured": "2023-08-07T20:44      Debitk.91 2099-12 Zahl.System VISA Debit",
                "internalTransactionId": "cfe5e62f2fbb2ea01f12bc965749d758"
            },
            {
                "entryReference": "REU5NjEyMDMwMDAwOTAwNTI5MDkwNC00Ny4xNTIwMjMtMDgtMDc=",
                "endToEndId": "483217578024369",
                "bookingDate": "2023-08-07",
                "valueDate": "2023-08-07",
                "transactionAmount": {
                    "amount": "-47.15",
                    "currency": "EUR"
                },
                "creditorName": "Amazon.de",
                "creditorAccount": {
                    "iban": "DE96120300009005290904"
                },
                "debtorAccount": {
                    "iban": "redacted"
                },
                "remittanceInformationUnstructured": "2023-08-06T13:25      Debitk.91 2099-12 Zahl.System VISA Debit",
                "internalTransactionId": "a2415cf5ecc258c1010d498212329446"
            },
            {
                "entryReference": "REU5NjEyMDMwMDAwOTAwNTI5MDkwNC0xMzkuOTAyMDIzLTA4LTA3",
                "endToEndId": "483217354487815",
                "bookingDate": "2023-08-07",
                "valueDate": "2023-08-07",
                "transactionAmount": {
                    "amount": "-139.90",
                    "currency": "EUR"
                },
                "creditorName": "TUEV SUED Auto Service",
                "creditorAccount": {
                    "iban": "DE96120300009005290904"
                },
                "debtorAccount": {
                    "iban": "redacted"
                },
                "remittanceInformationUnstructured": "2023-08-05T10:33      Debitk.91 2099-12 Zahl.System VISA Debit",
                "internalTransactionId": "84cb811266e18b717e8c13e91da6ef80"
            },
            {
                "entryReference": "REU5NjEyMDMwMDAwOTAwNTI5MDkwNC0zNS43MDIwMjMtMDgtMDc=",
                "endToEndId": "483216567592892",
                "bookingDate": "2023-08-07",
                "valueDate": "2023-08-07",
                "transactionAmount": {
                    "amount": "-35.70",
                    "currency": "EUR"
                },
                "creditorName": "Lieferando.de lieferser",
                "creditorAccount": {
                    "iban": "DE96120300009005290904"
                },
                "debtorAccount": {
                    "iban": "redacted"
                },
                "remittanceInformationUnstructured": "2023-08-04T18:27      Debitk.91 2099-12 Zahl.System VISA Debit",
                "internalTransactionId": "92ba427a910a77ec2d1a1830d718ef5c"
            },
            {
                "entryReference": "REU5NjEyMDMwMDAwOTAwNTI5MDkwNC0xNS4xMzIwMjMtMDgtMDQ=",
                "endToEndId": "483214423734087",
                "bookingDate": "2023-08-04",
                "valueDate": "2023-08-04",
                "transactionAmount": {
                    "amount": "-15.13",
                    "currency": "EUR"
                },
                "creditorName": "EDEKA ESSLINGER",
                "creditorAccount": {
                    "iban": "DE96120300009005290904"
                },
                "debtorAccount": {
                    "iban": "redacted"
                },
                "remittanceInformationUnstructured": "2023-08-02T12:40      Debitk.91 2099-12 Zahl.System VISA Debit",
                "internalTransactionId": "b2e47aba0b83105f19d7209f23b79f20"
            },

**Reply by Serph91P** [reply]
just tried to import 90 days it just keeps spamming the amazon account name. Which is obviously not correct.

**Reply by JC5** [reply]
It was the first one with that IBAN, so it keeps preferring that account.

**Reply by Serph91P** [reply]
but those are completely different creditor accounts ibans and names how can the importer just go to referencing this amazon account over and over. They have nothing to do with amazon and still this account gets taken.

**Reply by JC5** [reply]
They all seem to end on 904, right?

**Reply by Serph91P** [reply]
wtf you are right, so it is Nordigen whcih gets the data wrong. I was only looking at the names of the creditor and am like wtf the are completely different accounts like the supermarket and amazon and a car garage how are they the same creditor. Do you know if they have a support or something I can message maybe I can export a csv or something to get startet but automating the importer would be great.

**Reply by Serph91P** [reply, marked as answer]
ok sorry for the crazyness, found the error my bank seems to be not showing the iban where a transaction goes when you use ur card and nordigen is just putting a iban there which came before.

**Reply by JC5** [reply]
Not a problem, crazy indeed! Something for their support team I guess.

**Reply by Belgadon** [reply]
Have you found a way to import data from DKB in a useful manner?

**Comment by Serph91P** [comment]
> Have you found a way to import data from DKB in a useful manner?
